### DISCUSSION TITLE
When specifying names for classes, functions, etc, are string literals required

### DISCUSSION THREAD

**Discussion by tonygould** [pybind/pybind11#4944] [Q&A] [answered]
E.g. `py::class_` constructor
```
    class_(handle scope, const char *name, const Extra &...extra) {
```

adding a method

```
    template <typename Func, typename... Extra>
    class_ &def(const char *name_, Func &&f, const Extra &...extra) {
```

I'm not sure whether it would be ok to create a `std::string` on the fly and then pass in the `c_str()` pointer. The memory pointed to by the string would later become invalid -- is this ok? I.e., does pybind11 take a copy of the string passed in via the `const char*` parameters, or does it simply copy the pointer. It it copies just the pointer, I must ensure that memory to which the `const char*` points is immutable.

**Comment by evguran** [comment, marked as answer]
Hi, 
this name in the end comes to the function 
https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_FromString
which creates a new reference,
so you may not be bothered about memory state imho

**Comment by tonygould** [comment]
@evguran Many thanks, it's very kind of you to spend the time and answer this.
